### ISSUE TITLE
How is the best teacher model selected?

### ISSUE THREAD

**Issue by rimiyeyo** [etzinis/unsup_speech_enh_adaptation#5]
"Thank you for always answering kindly. I would like to ask how the teacher model's pretrained model is selected on 'comet.com'. Do you choose based on the highest mean value? Or do you choose based on the mos value? Please let me know.

**Comment by etzinis** [comment]
Hello there, the teacher model is just left to run for 100-or-so epochs or even simpler until convergence. To me, the point of unsupervised adaptation is that the pre-trained model on out-of-domain data should not be fine-tuned or selected using any in-domain validation set.

**Comment by rimiyeyo** [comment]
@etzinis Hi Thank you for reply. I ask the question again because it seems to be different from the intention of my question.

When I train the student model with the teacher model ckpt provided by you, it produces good results.
However, if I train the student model with the reproduced teacher model, trained by using the provided script, I get different results.

To sum up, the results are not reproducible. 
Are there any additional code changes or did you train the teacher with more epochs?
If not, are there any specific metrics you look at the comet.ml site?

always thanks
